pkg/maintenance/looseCoupling.go

```diff
@@ -0,0 +1,117 @@
+package maintenance
+
+import (
+	"context"
+	"fmt"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var waitForEtcdTimeout = 5 * time.Minute
+
+type looselyCoupledMaintenanceSwitch struct {
+	maintenanceModeSwitch
+	statefulSetClient statefulSetInterface
+	serviceInterface  serviceInterface
+}
+
+// NewWithLooseCoupling creates a switch that checks if the configuration registry (e.g., etcd) exists before switching.
+// If the registry does not exist, no switch is executed.
+func NewWithLooseCoupling(globalConfig globalConfig, clientSet statefulSetInterface, serviceInterface serviceInterface) *looselyCoupledMaintenanceSwitch {
+	return &looselyCoupledMaintenanceSwitch{
+		maintenanceModeSwitch: New(globalConfig),
+		statefulSetClient:     clientSet,
+		serviceInterface:      serviceInterface,
+	}
+}
+
+// ActivateMaintenanceMode activates the maintenance mode if the etcd exists and is ready.
+// This loose coupling enables us to perform restores on an empty cluster.
+func (lcms *looselyCoupledMaintenanceSwitch) ActivateMaintenanceMode(ctx context.Context, title string, text string) error {
+	if etcdReady, err := lcms.isEtcdReady(); err != nil {
```

Comment:
🚫 **[checkstyle]** <contextcheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function `isEtcdReady` should pass the context parameter